apm-agent-plugins/apm-micrometer-plugin/src/main/java/co/elastic/apm/agent/micrometer/MicrometerMetricsReporter.java

```diff
@@ -97,11 +99,21 @@ public void run() {
     }
 
     private static class MeterMapConsumer implements Consumer<Meter> {
+
+        private final List<WildcardMatcher> disabledMetrics;
+
+        public MeterMapConsumer(List<WildcardMatcher> disabledMetrics) {
+            this.disabledMetrics = disabledMetrics;
+        }
+
         final Map<Meter.Id, Meter> meters = new HashMap<>();
 
         @Override
         public void accept(Meter meter) {
-            meters.put(meter.getId(), meter);
+            Meter.Id meterId = meter.getId();
+            if (WildcardMatcher.anyMatch(disabledMetrics, meterId.getName()) == null) {
```

Comment:
I think compared to the actual serialization of the metrics the matching is not likely to be significant.

and idea to make it a bit more readable
```suggestion
            if (WildcardMatcher.isNoneMatch(disabledMetrics, meterId.getName())) {
```